File: formationJpaSpring/src/main/java/exoFormation/entity/Formateur.java
package exoFormation.entity;

import java.util.List;
import java.util.Set;

import javax.persistence.AttributeOverride;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.SequenceGenerator;
import javax.persistence.Table;

@Entity
@Table(name = "formateur")
@AttributeOverride(name = "adresse.numero", column = @Column(name = "formateur_numero", length = 50))
@AttributeOverride(name = "adresse.rue", column = @Column(name = "formateur_rue", length = 255))
@AttributeOverride(name = "adresse.codePostal", column = @Column(name = "formateur_cp", length = 50))
@AttributeOverride(name = "adresse.ville", column = @Column(name = "formateur_ville", length = 255))
@AttributeOverride(name = "id", column = @Column(name = "formateur_id"))
@AttributeOverride(name = "prenom", column = @Column(name = "formateur_prenom", length = 255, nullable = false))
@AttributeOverride(name = "nom", column = @Column(name = "formateur_nom", length = 255, nullable = false))
@AttributeOverride(name = "email", column = @Column(name = "formateur_email", length = 255))
@AttributeOverride(name = "telephone", column = @Column(name = "formateur_telephone", length = 20))
@SequenceGenerator(name = "seqPersonne", sequenceName = "formateur_seq", initialValue = 50, allocationSize = 1)
public class Formateur extends Personne {
	private boolean interne;
	private double cout;
	@OneToMany(mappedBy = "referent")
	private Set<Formation> formationsCommeReferent;

	@OneToMany(mappedBy = "formateur")
	private Set<ModuleFormation> modulesAnimes;

	public Formateur(String prenom, String nom, String email, String telephone, Adresse adresse, boolean interne,
			double cout) {
		super(prenom, nom, email, telephone, adresse);
		this.interne = interne;
		this.cout = cout;
	}

	public Formateur() {

	}

	public boolean isInterne() {
		return interne;
	}

	public void setInterne(boolean interne) {
		this.interne = interne;
	}

	public double getCout() {
		return cout;
	}

	public void setCout(double cout) {
		this.cout = cout;
	}

	public Set<Formation> getFormationsCommeReferent() {
		return formationsCommeReferent;
	}

	public void setFormationsCommeReferent(Set<Formation> formationsCommeReferent) {
		this.formationsCommeReferent = formationsCommeReferent;
	}

	public Set<ModuleFormation> getModulesAnimes() {
		return modulesAnimes;
	}

	public void setModulesAnimes(Set<ModuleFormation> modulesAnimes) {
		this.modulesAnimes = modulesAnimes;
	}

	public String getInfos() {
		return getId() + " " + getPrenom() + " " + getNom();
	}

}

File: angularFinal/src/app/model/formateur.ts
import { Adresse } from './adresse';
export class Formateur {
  public get cout(): number | undefined {
    return this._cout;
  }
  public set cout(value: number | undefined) {
    this._cout = value;
  }
  public get interne(): boolean | undefined {
    return this._interne;
  }
  public set interne(value: boolean | undefined) {
    this._interne = value;
  }
  public get adresse(): Adresse | undefined {
    return this._adresse;
  }
  public set adresse(value: Adresse | undefined) {
    this._adresse = value;
  }
  public get dtNaiss(): Date | undefined {
    return this._dtNaiss;
  }
  public set dtNaiss(value: Date | undefined) {
    this._dtNaiss = value;
  }
  public get email(): string | undefined {
    return this._email;
  }
  public set email(value: string | undefined) {
    this._email = value;
  }
  public get nom(): string | undefined {
    return this._nom;
  }
  public set nom(value: string | undefined) {
    this._nom = value;
  }
  public get prenom(): string | undefined {
    return this._prenom;
  }
  public set prenom(value: string | undefined) {
    this._prenom = value;
  }
  public get id(): number | undefined {
    return this._id;
  }
  public set id(value: number | undefined) {
    this._id = value;
  }
  constructor(
    private _id?: number,
    private _prenom?: string,
    private _nom?: string,
    private _email?: string,
    private _dtNaiss?: Date,
    private _adresse?: Adresse,
    private _interne?: boolean,
    private _cout?: number
  ) {}
}
